hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/lz4/Lz4Compressor.java

```diff
@@ -302,11 +303,20 @@ public synchronized long getBytesWritten() {
   public synchronized void end() {
   }
 
-  private native static void initIDs();
-
-  private native int compressBytesDirect();
-
-  private native int compressBytesDirectHC();
-
-  public native static String getLibraryName();
+  private int compressDirectBuf() {
```

Comment:
seems some of the methods in this class look exactly the same as in `SnappyCompressor` - perhaps we can do some refactoring later.

Reply:
Ok, currently this PR focuses on turning to use lz4-java. We can refactor compressor/decompressor classes later.